src/component/handlers/composite-android/DraftEditorCompositionHandlerAndroid.js

```diff
@@ -163,13 +169,19 @@ var DraftEditorCompositionHandlerAndroid = {
       handleMutations(mutationObserver.takeRecords());
     }
 
-    // If there are mutations now, restore the dom to prevent React from failing during reconciliation.
     if (hasMutation) {
+      // If there are mutations now, restore the dom to prevent React from failing during reconciliation.
       editor.restoreEditorDOM();
+    } else {
+      // If there are no mutations Draft is unaware of (deletions), flag it as native content.
+      compositionState = EditorState.set(compositionState, {
+        nativelyRenderedContent: compositionState.getCurrentContent(),
+      });
     }
 
     editor.setMode('edit');
     DraftEditorCompositionHandlerAndroid.commit(editor, compositionState);
+    cancelCompositionTimeout();
```

Comment:
< ⭐️ Should we do this before the commit since that's likely to kick off stuff that could take a while?

Reply:
Good point.
Since JS is single threaded, I don't think it matters, but I agree that it probably makes more sense earlier in the handler.